srk/src/syntax.ml

```diff
@@ -408,15 +408,15 @@ let substitute_sym srk subst sexpr =
           children
           Env.empty
       in
-      substitute srk (Env.find env) (subst k)
-      |> decapture srk 0 (depth - (List.length children))
+      let f (ind, typ) = try (Env.find env ind) with _ -> (mk_var srk ind typ) in
+      decapture srk (List.length children) (depth - (List.length children)) (subst k)
+      |> substitute srk f
```

Comment:
Style: better to avoid giving non-descriptive names to short functions, and generic error capture:
```
      |> substitute srk (fun (ind, typ) ->
            try Env.find env ind with Not_found -> mk_var srk ind typ)
```